src/OSPSuite.SimModelNative/src/FormulaChange.cpp

```diff
@@ -92,6 +92,8 @@ bool FormulaChange::PerformSwitchUpdate(double * y, double time)
 	//---- object to change is not a species - set new formula or value
 	if (_useAsValue)
 	{
+		if (_quantity->GetFormula())
+			switchJacobians = true;
```

Comment:
Here **switchJacobians** should be true only if the old formula depends on any state variable